drivers/w1/w1_ds2484.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "w1_ds248x.h"
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/zephyr.h>
+
+#define DT_DRV_COMPAT maxim_ds2484
+
+LOG_MODULE_REGISTER(ds2484, CONFIG_W1_LOG_LEVEL);
+
+struct ds2484_config {
+	struct w1_master_config w1_config;
+	const struct i2c_dt_spec i2c_spec;
+	const struct gpio_dt_spec slpz_spec;
+	bool apu;
+};
+
+struct ds2484_data {
+	struct w1_master_data w1_data;
+	uint8_t reg_device_config;
+};
+
+static int ds2484_reset_bus(const struct device *dev)
```

Comment:
The extracted functions take only the i2c device structure instead of the whole device; therefore they are pretty much needed.

Reply:
Sorry for the confusion, I implicitly meant if extracting the i2c device in the outer function was the best way to go. Because if you would pass the device you wouldn't need the wrapper for the DS2484, but would still need a wrapper for the channel selection in the case of a ds2482. So yeah, it's probably best to leave it as it is now, and have wrappers for drivers sharing the common ds248x functions.